Can't build on Ubuntu: error: ‘hash_combine’ is not a member of ‘boost’

Latest master: https://github.com/bigartm/bigartm/commit/e4bded2f65559fc9217170adcf7da6b6de33545c

```
romovpa@romovpa:~/bigartm/build$ make
[  1%] Built target gflags-static
[  3%] Built target google-glog
[ 14%] Built target libprotobuf
[ 25%] Built target libprotoc
[ 25%] Built target protoc
[ 47%] Built target libzmq
[ 70%] Built target libzmq-static
[ 74%] Built target rpcz
[ 78%] Built target rpcz_static
[ 80%] Built target protoc-gen-cpp_rpcz
[ 80%] Built target protoc-gen-python_rpcz
[ 80%] Building CXX object src/artm/CMakeFiles/artm-static.dir/core/processor.cc.o
In file included from /home/romovpa/bigartm/src/artm/../artm/core/processor.h:19:0,
                 from /home/romovpa/bigartm/src/artm/core/processor.cc:3:
/home/romovpa/bigartm/src/artm/../artm/core/common.h: In constructor ‘artm::core::CuckooWatch::CuckooWatch(std::string)’:
/home/romovpa/bigartm/src/artm/../artm/core/common.h:175:54: warning: ‘artm::core::CuckooWatch::start_’ will be initialized after [-Wreorder]
/home/romovpa/bigartm/src/artm/../artm/core/common.h:174:16: warning:   ‘artm::core::CuckooWatch* artm::core::CuckooWatch::parent_’ [-Wreorder]
/home/romovpa/bigartm/src/artm/../artm/core/common.h:150:12: warning:   when initialized here [-Wreorder]
/home/romovpa/bigartm/src/artm/../artm/core/common.h: In constructor ‘artm::core::CuckooWatch::CuckooWatch(std::string, artm::core::CuckooWatch*)’:
/home/romovpa/bigartm/src/artm/../artm/core/common.h:175:54: warning: ‘artm::core::CuckooWatch::start_’ will be initialized after [-Wreorder]
/home/romovpa/bigartm/src/artm/../artm/core/common.h:174:16: warning:   ‘artm::core::CuckooWatch* artm::core::CuckooWatch::parent_’ [-Wreorder]
/home/romovpa/bigartm/src/artm/../artm/core/common.h:152:3: warning:   when initialized here [-Wreorder]
/home/romovpa/bigartm/src/artm/core/processor.cc: In function ‘std::shared_ptr<artm::core::ModelIncrement> artm::core::InitializeModelIncrement(const artm::core::ProcessorInput&, const artm::ModelConfig&, const artm::core::TopicModel&)’:
/home/romovpa/bigartm/src/artm/core/processor.cc:78:17: warning: unused variable ‘counters’ [-Wunused-variable]
/home/romovpa/bigartm/src/artm/core/processor.cc: In function ‘std::shared_ptr<artm::utility::DenseMatrix<float> > artm::core::InitializeTheta(const artm::Batch&, const artm::ModelConfig&, const artm::core::DataLoaderCacheEntry*)’:
/home/romovpa/bigartm/src/artm/core/processor.cc:124:9: error: ‘hash_combine’ is not a member of ‘boost’
/home/romovpa/bigartm/src/artm/core/processor.cc:125:9: error: ‘hash_combine’ is not a member of ‘boost’
/home/romovpa/bigartm/src/artm/core/processor.cc: In function ‘void artm::core::InferThetaSparse(const artm::ModelConfig&, const artm::Batch&, const artm::core::InstanceSchema&, const artm::utility::CsrMatrix<float>&, const artm::core::TopicModel&, artm::utility::DenseMatrix<float>*, artm::utility::Blas*)’:
/home/romovpa/bigartm/src/artm/core/processor.cc:264:13: warning: unused variable ‘tokens_count’ [-Wunused-variable]
/home/romovpa/bigartm/src/artm/core/processor.cc: In function ‘void artm::core::UpdateNwtSparse(const artm::ModelConfig&, const artm::Batch&, const artm::core::Mask*, const artm::core::InstanceSchema&, const artm::utility::CsrMatrix<float>&, const artm::core::TopicModel&, const artm::utility::DenseMatrix<float>&, artm::core::ModelIncrement*, artm::utility::Blas*)’:
/home/romovpa/bigartm/src/artm/core/processor.cc:357:13: warning: unused variable ‘docs_count’ [-Wunused-variable]
make[2]: *** [src/artm/CMakeFiles/artm-static.dir/core/processor.cc.o] Error 1
make[1]: *** [src/artm/CMakeFiles/artm-static.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/artm/core/processor.cc
+++ b/src/artm/core/processor.cc
@@ -12,6 +12,7 @@
 #include "boost/lexical_cast.hpp"
 #include "boost/uuid/uuid_generators.hpp"
 #include "boost/uuid/uuid_io.hpp"
+#include "boost/functional/hash.hpp"
 
 #include "glog/logging.h"
 

--- a/src/artm_tests/repeatable_result_test.cc
+++ b/src/artm_tests/repeatable_result_test.cc
@@ -34,7 +34,7 @@ std::string runOfflineTest() {
   model_config.set_name(artm::test::Helpers::getUniqueString());
   artm::Model model(master_component, model_config);
 
-  std::vector<std::shared_ptr<::artm::Batch>> batches;
+  std::vector<std::shared_ptr< ::artm::Batch>> batches;
   int batches_size = 2;
   int nTokens = 10;
   srand(1);

